src/ddscxx/src/org/eclipse/cyclonedds/sub/qos/DataReaderQosDelegate.cpp

```diff
@@ -225,6 +236,7 @@ DataReaderQosDelegate::operator =(const org::eclipse::cyclonedds::topic::qos::To
     history_     = tqos.policy<dds::core::policy::History>();
     resources_   = tqos.policy<dds::core::policy::ResourceLimits>();
     ownership_   = tqos.policy<dds::core::policy::Ownership>();
+    datarepresentation_ = tqos.policy<dds::core::policy::DataRepresentation>();
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro. (BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).

Reply:
Will do.